eng/tools/generator/cmd/v2/common/cmdProcessor.go

```diff
@@ -19,11 +19,48 @@ import (
 func ExecuteGoGenerate(path string) error {
 	cmd := exec.Command("go", "generate")
 	cmd.Dir = path
-	output, err := cmd.CombinedOutput()
-	log.Printf("Result of `go generate` execution: \n%s", string(output))
+
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+
+	stderrPipe, err := cmd.StderrPipe()
 	if err != nil {
-		return fmt.Errorf("failed to execute `go generate` '%s': %+v", string(output), err)
+		return err
+	}
+
+	if err = cmd.Start(); err != nil {
+		return err
+	}
+
+	var stdoutBuffer bytes.Buffer
+	if _, err = io.Copy(&stdoutBuffer, stdoutPipe); err != nil {
+		return err
+	}
+
+	var stderrBuffer bytes.Buffer
+	if _, err = io.Copy(&stderrBuffer, stderrPipe); err != nil {
+		return err
+	}
+
+	fmt.Println(stdoutBuffer.String())
+	if stdoutBuffer.Len() > 0 {
+		if strings.Contains(stdoutBuffer.String(), "error   |") {
+			// find first error message until last
+			errMsgs := stdoutBuffer.Bytes()
+			index := regexp.MustCompile(`error   \|`).FindIndex(errMsgs)
+			if len(index) == 2 {
+				return fmt.Errorf("failed to execute `go generate`:\n%s", string(errMsgs[index[0]:]))
+			}
+		}
 	}
+
+	if err := cmd.Wait(); err != nil || stderrBuffer.Len() > 0 {
```

Comment:
Checking the error returning from cmd.Wait() is enough, can we removed `stderrBuffer.Len()>0`?

Reply:
When tsp compile, I encountered a non cmd error, but it has content in err buf

Reply:
The error returned by cmd is mainly `exit status 1`